cli/tests/activate.bats

```diff
@@ -3254,6 +3254,79 @@ EOF
 
 # ---------------------------------------------------------------------------- #
 
+# bats test_tags=activate,activate:attach
+# NB: There is a corresponding test in `services.bats`.
+@test "version: refuses to attach to an older activations.json version" {
+  project_setup
+
+  # TODO: Workaround for https://github.com/flox/flox/issues/2164
+  rm "${HOME}/.bashrc"
+
+  # Prevent backtraces from `flox-activations` leaking into output.
+  unset RUST_BACKTRACE
+
+  export -f jq_edit
+  FLOX_SHELL="bash" run "$FLOX_BIN" activate -- bash <(
+    cat << 'EOF'
+      echo "$PPID" > activation_pid
```

Comment:
Yeah we have to use PPID rather than $$ because we're exec'ing _flox_shell and passing that bash. I think that's fine
https://github.com/flox/flox/blob/27f968f8fb519ed753bf1cceb1a6be3920cc0ab7/assets/activation-scripts/activate.d/attach-command.bash#L34

Reply:
Yeah, realised the connection while working on https://github.com/flox/flox/pull/2380, which is kinda surprising in that:

- calling `activate -- bash` can actually invoke another shell like `zsh` first
- the PIDs we list are of an intermediate shell process rather than the command the user invoked

But both are beyond the scope of this PR.